pkg/repository/mirror.go

```diff
@@ -249,6 +326,71 @@ func (l *httpMirror) prepareURL(resource string) string {
 	return url
 }
 
+// Introduce implements the model.Model interface
+func (l *httpMirror) Introduce(id, name string, key *v1manifest.KeyInfo) error {
+	return errors.Errorf("introduce a fresher via the internet is not allowd, please set you mirror to a local one")
+}
+
+// Publish implements the model.Model interface
+func (l *httpMirror) Publish(manifest *v1manifest.Manifest, info model.ComponentInfo) error {
+	sid := uuid.New().String()
+
+	if info.Filename() != "" {
+		tarAddr := fmt.Sprintf("%s/api/v1/tarball/%s", l.Source(), sid)
+		resp, err := utils.PostFile(info, tarAddr, "file", info.Filename())
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode >= 300 {
```

Comment:
What about 301, 302, 303 and 307?

Reply:
It's unexpected, the server shouldn't return this.